src/extractor/edge_based_graph_factory.cpp

```diff
@@ -638,6 +595,475 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                                  << " turns over barriers";
 }
 
+// requires sorted candidates
+std::vector<EdgeBasedGraphFactory::TurnCandidate>
+EdgeBasedGraphFactory::optimizeCandidates(NodeID via_eid,
+                                          std::vector<TurnCandidate> turn_candidates) const
+{
+    BOOST_ASSERT_MSG(std::is_sorted(turn_candidates.begin(), turn_candidates.end(),
+                                    [](const TurnCandidate &left, const TurnCandidate &right)
+                                    {
+                                        return left.angle < right.angle;
+                                    }),
+                     "Turn Candidates not sorted by angle.");
+    if (turn_candidates.size() <= 1)
+        return turn_candidates;
+
+    const auto getLeft = [&turn_candidates](std::size_t index)
+    {
+        return (index + 1) % turn_candidates.size();
+    };
+    const auto getRight = [&turn_candidates](std::size_t index)
+    {
+        return (index + turn_candidates.size() - 1) % turn_candidates.size();
+    };
+
+    // handle availability of multiple u-turns (e.g. street with separated small parking roads)
+    if (turn_candidates[0].instruction == TurnInstruction::UTurn && turn_candidates[0].angle == 0 &&
+        turn_candidates.size() > 1)
+    {
+        if (turn_candidates[getLeft(0)].instruction == TurnInstruction::UTurn)
+            turn_candidates[getLeft(0)].instruction = TurnInstruction::TurnSharpLeft;
+        if (turn_candidates[getRight(0)].instruction == TurnInstruction::UTurn)
+            turn_candidates[getRight(0)].instruction = TurnInstruction::TurnSharpRight;
+    }
+
+    const auto keepStraight = [](double angle)
+    {
+        return std::abs(angle - 180) < 5;
+    };
+
+    if (turn_candidates.size() > 1)
```

Comment:
Change to early-exit style `if (turn_candidate <= 1) { return turn_candidates; } BOOST_ASSERT(turn_candidates.size() > 1);` to circumvent the indentation depth.